src/java.base/share/classes/java/net/SocketPermission.java

```diff
@@ -984,15 +992,14 @@ boolean impliesIgnoreMask(SocketPermission that) {
 
                 // XXX: if all else fails, compare hostnames?
                 // Do we really want this?
-                if (this.cname == null) {
+                if (this.cnames == null) {
                     this.getCanonName();
                 }
 
-                if (that.cname == null) {
+                if (that.cnames == null) {
                     that.getCanonName();
                 }
-
-                return (this.cname.equalsIgnoreCase(that.cname));
+                return this.cnames[0].equalsIgnoreCase(that.cnames[0]);
```

Comment:
@jaysk1 On the event of us reaching here, won't `this` and `that` have multiple canonical names each, in different orders? Do you think we need to compare the arrays here?